### ISSUE TITLE
NotImplementedError for MSC Nastran v2021.4

### ISSUE THREAD

**Issue by fmamitrotta** [SteveDoyle2/pyNastran#682]
# Summary
I was testing the latest MSC Nastran version that I got, v2021.4 (previously I was using v2018.1) and I came across a `NotImplementedError` when I tried to read an op2 file with the `read_op2` function. The `NotImplementedError` came out from the `op2_reader.py` module, where I found the following variable that presumably containts the supported versions of MSC Nastran:
```
MSC_LONG_VERSION = [
    b'XXXXXXXX20140', b'XXXXXXXX20141', b'XXXXXXXX20142',
    b'XXXXXXXX20150', b'XXXXXXXX20151', b'XXXXXXXX20152',
    b'XXXXXXXX20160', b'XXXXXXXX20161', b'XXXXXXXX20162',
    b'XXXXXXXX20170', b'XXXXXXXX20171', b'XXXXXXXX20172',
    b'XXXXXXXX20180', b'XXXXXXXX20181', b'XXXXXXXX20182',
    b'XXXXXXXX20190', b'XXXXXXXX20191', b'XXXXXXXX20192',
    b'XXXXXXXX20200', b'XXXXXXXX20201', b'XXXXXXXX20202',
    b'XXXXXXXX20210', b'XXXXXXXX20211', b'XXXXXXXX20212',
    b'XXXXXXXX20220', b'XXXXXXXX20221', b'XXXXXXXX20222',
]
```

Is there a reason why only the versions .0, .1 and .2 of the various MSC Nastran editions are included?

I tried to add `b'XXXXXXXX20214'` to the list and the python script seemed to work, however I got the following warning which before was not appearing for Nastran v2018.1:
```
WARNING: subcase.py:683 nwords_to_lsem=1000 nwords_to_lsem//4=250
```

Is there anything wrong with using v2021.4?

# Versions
`platform.python_version()` --> 3.9.7
`pyNastran.__version__` --> 1.4.0+dev.3b82e5518

**Comment by SteveDoyle2** [comment]
> Is there a reason why only the versions .0, .1 and .2 of the various MSC
Nastran editions are included?

As far as I know that's all that existed.  They changed their version
numbering at some point.  It used to only be XXXXXXXX, which is kind of
hilarious.

That warning is related to the case control deck parsing that is weird in
the newer versions.  They keep adding fields to the case control deck,
which makes coming up with a general routine difficult.

> Is there anything wrong with using v2021.4?

Not that I know of.  I know they changed some stuff in 2021.x regarding the
RBE1/2/3, which is fixed, but buyer beware.

**Comment by fmamitrotta** [comment]
Ok understood! Do you want me to add v2021.3 and 2021.4 to `MSC_LONG_VERSION` and make a pull request? Is there something similar in other modules?

Regarding the warning, so basically that shows up only in newer versions, correct? It might explain why I did not see it with v2018.1.

**Comment by SteveDoyle2** [comment]
A pull request would be fine.  

> Is there something similar in other modules?

I don't follow.  The op2 is the only thing that cares about versioning.  It's mainly used to help process the geometry.  

For example, there are 2 versions of the MSC CQUAD4 for SOL 101 and the only difference is there are 2 more zero placeholder values for something and a -1 at the end.  They have the same key code (there are 3 integers that flag that you have a GRID, CQUAD4, etc.), but they have a different format.  You end up looking at the factors.  Let's say you have 1000 values and something that is either 10 or 12 fields long, so find the integer N in 12*N=1000 or 10*N=1000.  That's easy, but what if it's 12*N=1200 and 10*N=1200?  You either have 100 elements or 120 elements.  You can look at things like node (GRID) , element (CQUAD4), property (PSHELL), material id  (MAT1) and verify they're always positive, but there are internal ids that don't follow the convention.

> Regarding the warning, so basically that shows up only in newer versions, correct? It might explain why I did not see it with v2018.1.

Yes.  I don't know when they added it.  I have access to MSC 2005 and MSC 2020 and a few models from in-between years that are tests.

**Comment by fmamitrotta** [comment]
I have made the pull request! I have only added v2021.4 because I don't know for sure whether v2021.3 exists, it was just an assumption from my side (probably it is true but they could also have skipped it).

I did not know that op2 is the only thing that cares about versioning, I was asking about other modules for this reason. Thanks for the explanation! I could not follow all the details but I got your point.
